### ISSUE TITLE
HighLight Drab & Drop Behaviour

### ISSUE THREAD

**Issue by Thaerious** [Thaerious/NERVE#91]
There is an inconsistent / un-intuitive behavior with how drag/drop and highlighting work together.   There are a number of considerations when building in this behavior.  This ticket is to discuss how/when highlighting should start and end.

**Comment by abramoffcwrc** [comment]
Can you explain what you mean by how/when highlighting should start and end?

**Comment by Thaerious** [comment]
The following is an example of what I'm getting at for how to describe behavior:

_When you click on an entity it turns yellow indicating it is highlighted, when you click on a different entity, without pressing any keys, the first entity is no longer highlighted, the second entity becomes highlighted._ 

This is a specification of program behavior which I can code to, as opposed to saying something like "highlighting doesn't work".  Hope that clears things up.

**Comment by SusanBrown** [comment]
Sounds like a very good thing to try to articulate precisely. 

Anita or Kristi, could one of you try to draft comprehensively what the expected behaviour should be?

Ed would it be helpful to mention ways in which it is not doing what we expect or do you want to keep this focused on the desiderata?

**Comment by Thaerious** [comment]
> Ed would it be helpful to mention ways in which it is not doing what we expect 

Yes as far as it's a specification, I intend to rework the way highlighting works according to how the end user would expect it to.  This isn't so much a bug fix as a rework where we will start with a proper specification.  Not everything will go this way due to time constraints, but I'd like to do as much like this as is possible.  It's a lot more concise and proper.

**Comment by abramoffcwrc** [comment]
I'm thinking that if you click an entity, it should highlight. If you ctrl-click on another, it should select both, and so on. If you shift click on two at different places on the page, it should highlight all in between. 

Right now you can highlight multiple entities by pressing shift, and this is counter-intuitive.

**Comment by Thaerious** [comment]
Bug:  selected text, context menu open -> causes unselect

**Comment by abramoffcwrc** [comment]
I just tested this today and it seems the same as last time.CTRL-click should select as many different tags as you click on, and SHIFT-click should select all the tags in between where you click.
